### PR TITLE
Payment soft delete

### PR DESCRIPTION
**Pull request by GabrielPradoAsaas**
### Impacto
Classe Payment herdando de BaseEntity e soft delete implementado

### PR Predecessora
- https://github.com/pedrooalves/goCharges/pull/9

### CODE REVIEW

File: grails-app/services/gocharges/PaymentService.groovy [hunk 1]
```diff
@@ -58,16 +60,14 @@ class PaymentService {
...
-
     public Payment findById(Long id) {
         return Payment.get(id)
     }
+
+    public void delete(Long id) {
+        Payment payment = Payment.get(id)
+        payment.deleted = true
+
+        payment.save()
```
Comment (Kpotz) [inline grails-app/services/gocharges/PaymentService.groovy:71]:
```suggestion
        payment.save(failOnError: true)
```

File: grails-app/controllers/gocharges/PaymentController.groovy [hunk 2]
```diff
@@ -58,4 +58,12 @@ class PaymentController {
     public showForm() {
         chain(action: "index", model: [showNewPaymentForm: true])
     }
+
+    public delete() {
```
Comment (jpdev01) [inline grails-app/controllers/gocharges/PaymentController.groovy:62]: Qual é o tipo de retorno?

File: grails-app/controllers/gocharges/PaymentController.groovy [hunk 3]
```diff
@@ -55,7 +55,15 @@ class PaymentController {
         }
     }
 
-    public showForm() {
+    public Object showForm() {
```
Comment (jpdev01) [inline grails-app/controllers/gocharges/PaymentController.groovy:58]: Não seria Map?

**Review by jpdev01 — APPROVED** [review]

**Review by Kpotz — APPROVED** [review]

**Comment by Kpotz** [comment]
Resolver conflitos

**Review by jpdev01 — APPROVED** [review]

**Comment by GabrielPradoAsaas** [comment]
> Resolver conflitos

Resolvidos

**Review by Kpotz — APPROVED** [review]

**Pull request merged**